routing_common/bicycle_model.cpp

```diff
@@ -166,11 +166,13 @@ VehicleModel::LimitsInitList UkraineOptions()
 }
 
 VehicleModel::SurfaceInitList const kBicycleSurface = {
-  // {{surfaceType, surfaceType}, {weightFactor, etaFactor}}
+  // {{surfaceType}, {weightFactor, etaFactor}}
   {{"psurface", "paved_good"}, {1.0, 1.0}},
   {{"psurface", "paved_bad"}, {0.8, 0.8}},
   {{"psurface", "unpaved_good"}, {1.0, 1.0}},
-  {{"psurface", "unpaved_bad"}, {0.3, 0.3}}
+  {{"psurface", "unpaved_bad"}, {0.3, 0.3}},
+  // no dedicated cycleway, doesn't mean that bicycle is not allowed, just lower weight
+  {{"hwtag", "nocycleway"}, {0.8, 0.8}},
```

Comment:
Does it mean that `highway=secondary cycleway=no` is considered worse/slower than "plain" `highway=secondary`?
Why? Because of a chance that the latter might have an untagged cycleway?

Reply:
Yes. ```highway=secondary AND cycleway=no``` is worth for bicycle than plain ```highway=secondary```
Maybe a bit controversial, but we had claims, and it was the best option-to-fix, AFAIR.

Reply:
Yeap it sounds very controversial. Are there any tests or links to users' reports/claims? (I didn't find)

I'd understand a very slight preference (like 0.95) but 20% is a lot.

Reply:
I found this case: https://github.com/orgs/organicmaps/discussions/5158#discussioncomment-11125829

Though it looks more like a coincidence rather than something to warrant this penalty.